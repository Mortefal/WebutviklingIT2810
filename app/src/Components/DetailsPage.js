import React from 'react';
import Button from '@material-ui/core/Button';
import Modal from '@material-ui/core/Modal';
import Grid from "@material-ui/core/Grid/Grid";
import ButtonBase from "@material-ui/core/ButtonBase/ButtonBase";
import BottleWine from "@material-ui/core/SvgIcon/SvgIcon";
import Typography from "@material-ui/core/Typography/Typography";
import {withStyles} from "@material-ui/core";
import PropTypes from 'prop-types';

const styles = theme => ({
    paper: {
        position: 'relative',
        width: theme.spacing.unit * 50,
        backgroundColor: theme.palette.background.paper,
        boxShadow: theme.shadows[5],
        padding: theme.spacing.unit * 4,
        top: 200,
        left:200
    },
    image: {
        width: 128,
        height: 128,
        /* Tror det skal være mulig å hente inn bilde ved å ta response.images[2].url hvor formatet er thumbnail */
    },
});

class DetailsPage extends React.Component{
    constructor(props) {
        super(props);
        this.state = ({
            openModal: false,
        })
    }

    handleOpen = () =>{
        this.setState({
            openModal: true
        })
    };

    handleClose= () =>{
        this.setState({
            openModal: false
        })
    };

    render() {
        const{classes, title, description} = this.props;
        return (
            <div>
                <Button className="modalButton" onClick={this.handleOpen}>Mer Info</Button>
                <Grid>
                    <div>
                        <Modal
                            aria-labelledby="simple-modal-title"
                            aria-describedby="simple-modal-description"
                            open={this.state.openModal}
                            onClose={this.handleClose}
                        >
                            <div className={classes.paper}>
                                <Grid>
                                    <ButtonBase className={classes.image}>
                                        <BottleWine style={{fontSize: 128}}/>
                                    </ButtonBase>
                                    <Typography gutterBottom variant="column">{title}</Typography>
                                    <Typography gutterBottom >{description}</Typography>
                                </Grid>
                            </div>
                        </Modal>
                    </div>
                </Grid>
            </div>
        );
    }

}

DetailsPage.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(DetailsPage);