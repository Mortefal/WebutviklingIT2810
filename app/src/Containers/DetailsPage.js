import React from 'react';
import Button from '@material-ui/core/Button';
import Modal from '@material-ui/core/Modal';
import Grid from "@material-ui/core/Grid/Grid";
import ButtonBase from "@material-ui/core/ButtonBase/ButtonBase";
import BottleWine from 'mdi-material-ui/BottleWine';
import Typography from "@material-ui/core/Typography/Typography";
import {withStyles} from "@material-ui/core";
import PropTypes from 'prop-types';
import FavoriteHeart from "../Components/FavoriteHeart";
import {hideInfo, showInfo} from "../Actions/actions";
import connect from "react-redux/es/connect/connect";
import rootReducer from "../Reducers/reducers";

const styles = theme => ({
    paper: {
        position: 'absolute',
        width: theme.spacing.unit * 50,
        backgroundColor: theme.palette.background.paper,
        boxShadow: theme.shadows[5],
        padding: theme.spacing.unit * 4,
        top: 100,
        left: 500,
        right: 0,
        margin: 0,
    },
    image: {
        width: 256,
        height: 256,
        justify: "flex-start",
        alignItems: "flex-start",
        borderBottomColor: '#ddd',
        borderBottomWidth: 2,
    },
    sideBox3: {
        height: 40,
        width:80,
    },
    sideBox1: {
        height: 100,
        width: 80,
    },
    sideBox2: {
        height: 40,
        width: 40,
    },
});
class DetailsPage extends React.Component {
    constructor(props) {
        super(props);
        this.show = this.show.bind(this);
        this.hide = this.hide.bind(this);
    }

    show(e){
        this.props.showInfo(e)
    }
    hide(e){
        this.props.hideInfo(e)
    }
    render() {
        const {classes, title, isFav, aroma, country, taste, abv, openModal} = this.props;
        return (
                <div>
                    <Button className="modalButton" onClick={(e) =>{this.props.show(e)}} >Mer Info</Button>
                    <Grid>
                        <div>
                            <Modal
                                aria-labelledby="simple-modal-title"
                                aria-describedby="simple-modal-description"
                                open={openModal}
                                onClose={(e) => {this.props.hide(e)}}
                                style={{alignItems: 'center', justifyContent: 'center'}}
                            >
                                <div className={classes.paper}>
                                    <Grid>
                                        <Grid item container direction={"row-reverse"}><FavoriteHeart isFav={isFav}/></Grid>
                                        <Grid container spacing={16} alignItems={"center"} alignContent={"center"}>
                                            <Grid item xs={6}>
                                                <ButtonBase className={classes.image}>
                                                    <BottleWine style={{fontSize: 256}}/>
                                                </ButtonBase>
                                            </Grid>

                                            <Grid item xs={6}>

                                                <Grid direction={"column"} justify={"center"} container item xs={12} alignItems={"center"}>
                                                    <Grid item xs={4} className={classes.sideBox3} >
                                                        Opprinnelsesland: {country}
                                                    </Grid>
                                                    <Grid className={classes.sideBox3} item xs={4}>
                                                        {abv}%
                                                    </Grid>
                                                </Grid>
                                            </Grid>
                                        </Grid>
                                    </Grid>
                                    <Grid>
                                        <Typography gutterBottom variant="h5">{title}</Typography>
                                        <Typography gutterBottom variant="body1">{aroma}{taste}</Typography>
                                    </Grid>
                                </div>
                            </Modal>
                        </div>
                    </Grid>
                </div>
            );
        }

}
const mapStateToProps = state => {
    return({
        openModal: state.displayInfo.openModal,
    })

}
const mapDispatchToProps = dispatch => {
    return {
        show: e => {
            dispatch(showInfo(e))
        },
        hide: e => {
            dispatch(hideInfo(e))
        }
    }
}


DetailsPage.propTypes = {
classes: PropTypes.object.isRequired,
};
export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(DetailsPage));
