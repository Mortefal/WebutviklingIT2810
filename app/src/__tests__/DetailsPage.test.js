import React from 'react';
import DetailsPage from '../Containers/DetailsPage';
import renderer from 'react-test-renderer'


describe('Snapshot test of DetailsPage', ()=> {
    it('renders correctly', () => {
        const tree = renderer
            .create(<DetailsPage title="Test" isFav="True" aroma="rÃ¸yk" country="England" taste="noe" abv={12}/>)
            .toJSON();
        expect(tree).toMatchSnapshot()
    })
});
