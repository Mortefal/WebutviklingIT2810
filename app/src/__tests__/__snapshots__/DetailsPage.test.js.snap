// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of DetailsPage renders correctly 1`] = `
<div>
  <button
    className="MuiButtonBase-root-32 MuiButton-root-6 MuiButton-text-8 MuiButton-flat-11 modalButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-7"
    >
      Mer Info
    </span>
    <span
      className="MuiTouchRipple-root-134"
    />
  </button>
  <div
    className=""
  >
    <div />
  </div>
</div>
`;
